crates/deltalake-core/src/operations/vacuum.rs

```diff
@@ -195,12 +195,48 @@ impl VacuumBuilder {
             .ok_or(DeltaTableError::NoMetadata)?
             .partition_columns;
 
+        let managed_files = self
+            .snapshot
+            .files()
+            .iter()
+            .map(|a| a.path.clone())
+            .chain(
+                self.snapshot
+                    .all_tombstones()
+                    .iter()
+                    .map(|r| r.path.clone()),
+            )
+            .chain(self.snapshot.files().iter().filter_map(|a| {
+                return if let Some(deletion_vector) = &a.deletion_vector {
+                    if let Ok(parent) = &Url::parse(self.log_store.root_uri().as_str()) {
+                        if let Ok(dv_absolut_path) = deletion_vector.absolute_path(&parent) {
+                            Some(dv_absolut_path?.to_string())
+                        } else {
+                            None
+                        }
+                    } else {
+                        None
+                    }
+                } else {
+                    None
+                };
+            }))
+            .collect::<HashSet<String>>();
+
         while let Some(obj_meta) = all_files.next().await {
             // TODO should we allow NotFound here in case we have a temporary commit file in the list
             let obj_meta = obj_meta.map_err(DeltaTableError::from)?;
-            if valid_files.contains(&obj_meta.location) // file is still being tracked in table
-            || !expired_tombstones.contains(obj_meta.location.as_ref()) // file is not an expired tombstone
-            || is_hidden_directory(partition_columns, &obj_meta.location)?
+
+            if is_hidden_file(partition_columns, &obj_meta.location)? {
```

Comment:
I renamed this to is_hidden_file since we are not only ignoring directories but also single files